app/models/post.rb

```diff
@@ -1,6 +1,14 @@
 class Post < ApplicationRecord
-  has_many :comments
-  has_many :likes
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
 
-  belongs_to :users
+  belongs_to :user, foreign_key: 'author_id'
+
+  def self.count_posts(user)
+    user.update(postCounter: user.posts.size)
+  end
+
```

Comment:
- [Optional] Dear @khanh-devos, for such a method which is expected to run every time we create a new post it's a better practice to apply the `after_save` helper. This way we can keep our data consistent and we don't need to call the `count_posts` method to update the author post counter, This also goes for all the other methods inside the `Like` and `Comment` controllers.  [see here for more](https://guides.rubyonrails.org/active_record_callbacks.html)